src/lwc/bdiObjectMappings/bdiObjectMappings.js

```diff
@@ -176,20 +152,25 @@ export default class bdiObjectMappings extends LightningElement {
             });
     }
 
-    /*******************************************************************************
-    * @description Call apex method 'getObjectOptions' to get
-    * a list of all objects that will be valid for creating object mappings on.
-    */
-    retrieveObjectOptions() {
-        getObjectOptions()
-            .then(result => {
-                this.objectOptions = result;
-            })
-            .catch(error => {
-                this.error = error;
-                this.handleError(error);
-            });
+    /**
+     * @description Extracts broken field mapping references from retrieved
+     * object mappings
+     * @param objectMappings
+     */
+    processBrokenMappingReferences (objectMappings) {
+        this.brokenMappings = [];
+        objectMappings.forEach(mapping => {
+            if (mapping.hasOwnProperty('Field_Mappings')) {
+                mapping.Field_Mappings.forEach(fieldMapping => {
+                    if (fieldMapping.Is_Broken && !fieldMapping.Is_Deleted) {
```

Comment:
Definitely see some overlap here with the other processBrokenFieldMappingReferences function but maybe we can't avoid it ...